serializers/api/vm.rb

```diff
@@ -13,7 +13,8 @@ def self.base(vm)
       unix_user: vm.unix_user,
       ip6: vm.ephemeral_net6&.nth(2),
       ip4: vm.ephemeral_net4,
-      projects: Serializers::Api::Project.serialize(vm.projects)
+      nics: vm.nics.map { |nic| Serializers::Api::Nic.serialize(nic) },
+      firewalls: vm.firewalls.map { |fw| Serializers::Api::Firewall.serialize(fw) }
```

Comment:
Do we plan to include associated resources in the vm body? It's a general design question.
- We have `POST vm/firewall-rule` and `DELETE vm/firewall-rule/ubid` why don't we have `GET vm/firewall-rule` to get the list of firewall rules?
- If we are planning to include in the single vm payload but not in the vm list payload, I suggest to divide serializer to two structures. example: https://github.com/ubicloud/ubicloud/blob/main/serializers/web/vm.rb#L25
- If we are planning to include in the vm list payload, I highly recommend to using eager loading while getting the list of virtual machines. If we don't use eager loading the vm listing will send lots of DB query.
```
TOTAL QUERY COUNT
- Get vms: 1
- Get nic for each vm: Vm count
- Get private subnet for each nic: Nic count
- Get firewall for each vm: Vm count
- Get firewall rules for each firewall rule: Firewall count
```

Reply:
Divided serializer into 2